script/mafia

```diff
@@ -26,16 +26,23 @@ run_upgrade () {
 
   trap clean_up EXIT
 
+  MAFIA_CUR="$0"
+
+  if [ -L "$MAFIA_CUR" ]; then
+    echo 'Refusing to overwrite a symlink; run `upgrade` from the canonical path.' >&2
```

Comment:
```
file exists, but not symlink = no output, exit code 1
```

That is the problematic case. On linoox it return 0 and prints canonical path if it is a directory.